monkey/infection_monkey/i_puppet/target_host.py

```diff
@@ -10,6 +10,10 @@
 from . import PortScanData
 
 
+# TODO: do we really need this? Can't we just have
+# `TargetHost.ports_status` be `Dict[NetworkPort, PortScanData]`?
+# `PortScanData` already has protocol information so having `tcp_ports`
+# and `udp_ports` is redundant.
```

Comment:
@mssalvatore

Reply:
Yes. Consider it an index. If I want to only consider TCP or UDP ports, it's quick and easy for me to do so; I don't need to loop through all of the data and filter. Simultaneously `PortScanData` objects contain all of the data necessary for them to be meaningful.